apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx

```diff
@@ -105,8 +110,38 @@ class VisualizerModal extends React.Component {
     return columns.filter(column => isColumnNumeric(records, column));
   });
 
+  getValuesForFilterColumn = memoize((records, column) => {
+    if (!records || !column) {
+      return [];
+    }
+    let values = Array.from(new Set(records.map(record => record[column])));
+    return sortAlphabeticallyOrNumerically(values);
+  });
+
+  filterRecords = memoize((records, column, value) => {
+    if (!records) {
+      return [];
+    }
+    if (!column || !value) {
```

Comment:
If you want to be able to filter to JS falsy values like zero and empty string, you'll want to change this check to something more specific like `value === undefined`.